include/xsimd/types/xsimd_api.hpp

```diff
@@ -1431,6 +1461,21 @@ namespace xsimd
         return kernel::round<A>(x, A {});
     }
 
+    /**
+     * @ingroup batch_rounding
+     *
+     * Computes the batch of nearest integer values to scalars in \c x (in
+     * floating point format), rounding halfway cases to even.
+     * @param x batch of floating point values.
+     * @return the batch of nearest even integer values.
+     */
+    template <class T, class A>
+    inline batch<as_integer<T>, A>
+    round_to_even(batch<T, A> const& x) noexcept
+    {
+        return kernel::round_to_even(x, A {});
```

Comment:
That's a surprising function, I'm curious about its usage?

Reply:
I think I made a mistake with the naming there ^^

The purpose of this function is to round floating-point numbers to integer-typed integers. Currently, `to_int` truncates (cvttps), and `round` makes an horrifying number of compares, moves, and blends, and returns a batch of floating points. See https://godbolt.org/z/q4rc6dvW3.